test/gateway/index.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+const API = require('../../src/http')
+const ncp = require('ncp').ncp
+const path = require('path')
+const clean = require('../utils/clean')
+
+describe('HTTP GATEWAY', () => {
+  const repoExample = path.join(__dirname, '../go-ipfs-repo')
+  const repoTests = path.join(__dirname, '../repo-tests-run')
+
+  let http = {}
+  let gateway
+
+  before((done) => {
+    http.api = new API(repoTests)
+
+    ncp(repoExample, repoTests, (err) => {
+      expect(err).to.not.exist()
+
+      http.api.start(false, () => {
+        gateway = http.api.server.select('Gateway')
+        done()
+      })
+    })
+  })
+
+  after((done) => {
+    http.api.stop((err) => {
+      expect(err).to.not.exist()
+      clean(repoTests)
+      done()
+    })
+  })
```

Comment:
Does the gateway need the go-ipfs test repo? It would be better to see it working with the default js-ipfs repo.